### PR TITLE
rework on search bar and questions components. 

### PR DESCRIPTION
**Pull request by sunkim0330**
@AlexArms @kate-darby @chrimack Hey all, after looking at other people's QandA section, I realized I was missing a lot of functionalities. So, I reworked on search bar and questions components. I basically combined them and now I can filter questions (with answers!) properly.  I am still working on my styling.

### CODE REVIEW

File: client/src/Q&As/SearchQandA.jsx [hunk 1]
```diff
@@ -22,38 +21,58 @@ const Questions = ({ product, setDateFormat }) => {
...
-  }, []);
-
-  const loadQuestions = questions.slice(0, loadPage).map((question) => {
+  const filteredQuestion = questions.filter((val) => {
+    if (search === '' || search.length < 3 ) {
+      return val;
+    } else if (val.question_body.toLowerCase().includes(search.toLowerCase())) {
+      return val;
+    }
+  }).slice(0, loadPage).map((question, key) => {
```
Comment (chrimack) [inline client/src/Q&As/SearchQandA.jsx:37]: Nice chaining going on here

File: client/src/Q&As/SearchQandA.jsx [hunk 2]
```diff
@@ -22,38 +21,58 @@ const Questions = ({ product, setDateFormat }) => {
...
     return (
       <div className="questions_div" key={question.question_id}>
         <Styles.QuestionsContainer>
-          Q: {question.question_body} <Styles.addAnswerButton onClick={() => setShow(true)} >Add Answer</Styles.addAnswerButton>
-          <Modal title="Submit Your Answer" subTitle={product.name} id={question.question_id} questionBody={question.question_body} show={show} onClose={() => setShow(false)}/><Helpful origin="qa/questions" id={question.question_id} helpCount={question.question_helpfulness}/><br/>
+          Q: {question.question_body} <Styles.addAnswerButton onClick={() => setShow(true)} >Add Answer</Styles.addAnswerButton><Helpful origin="qa/questions" id={question.question_id} helpCount={question.question_helpfulness}/>
+          <Modal title="Submit Your Answer" subTitle={product.name} id={question.question_id} questionBody={question.question_body} show={show} onClose={() => setShow(false)}/><br/>
```
Comment (chrimack) [inline client/src/Q&As/SearchQandA.jsx]: Consider adding new lines to make 41 and 42 more readable

**Review by chrimack — COMMENTED** [review]
Nice work with the search bar, it works great!

I pulled the code and tested it. After I clicked the 'MORE ANSWERED QUESTIONS' button a few times there was an error from Answers get request. Not sure why that only happens after 3 or 4 clicks.

The CSS is looking really good now. I'd suggest setting the height so that it doesn't keep changing when you search. Also you could make it scrollable so that it doesn't keep getting bigger when you add more answers.

File: client/src/Q&As/SearchQandA.jsx (hunk 2 above)
Comment (sunkim0330) [reply client/src/Q&As/SearchQandA.jsx]: fixed it! :)

**Comment by sunkim0330** [comment]
I also added scroll in index style as a temporary fix. It's not doing what I want it to do right now, but it's not going over my grid border

File: client/src/Q&As/Styles.js [hunk 3]
```diff
@@ -3,14 +3,61 @@ import styled, {css} from 'styled-components';
...
+  gap: 10px 0px;
+  grid-template-areas:
+    "search ."
+    "question ."
+    "buttons-buttons .";
+  color: #6f67bf;
+`;
+
+export const QuestionsContainer = styled.div.attrs({className: 'Questions_container'})`
+  overflow: scroll;
```
Comment (chrimack) [inline client/src/Q&As/Styles.js]: This line makes every question scrollable. You can create a div that holds all of the questions/answers, then make that scrollable. That way only the q/as move and the search bar and buttons will stay visible.

Comment (sunkim0330) [reply client/src/Q&As/Styles.js]: ah okay I'll try that! I'm still working on it 😖

**Pull request merged**